woocommerce/woocommerce
# CES exit prompt for product editing screens

### All Submissions:






### Changes proposed in this Pull Request:

This add's initial logic to make it easy to track if users leave pages with un-saved changes, for showing a feedback notice.
This would only be enabled if the user has enabled tracking.

Partially addresses #35126  .

https://user-images.githubusercontent.com/2240960/203960252-d8a73358-ec2b-405f-a953-55819956152f.mp4

### How to test the changes in this Pull Request:



1. Load this branch, build it, and make sure you have the `new-product-management-experience` feature flag enabled (you can do so using the latest version of the Beta tester within the mono repo).
2. Make sure you have tracks enabled and also outputted in the console: localStorage.setItem( 'debug', 'wc-admin:*' );
3. Go to **Product > Add New** and make some changes
4. Now go to another page like **WooCommerce > Home** and select **Leave** when it warns you about unsaved changes.
5. A notice should show on the new page that allows you to share feedback, something to the affect of `We noticed you started editing a product, then left....`
6. Click the share feedback and submit feedback, notice how the `wcadmin_ces_snackbar_view`, `wcadmin_ces_view`, and `wcadmin_ces_feedback` include the `ces_location` prop with the  `outside` value.
You may also want to try dismissing the notice and see if the prop has been added to `wcadmin_ces_snackbar_dismiss`
7. Repeat steps 4-6 for all these pages: Product edit page, New product MVP page, edit product MVP page.
8. Now delete the `woocommerce_ces_shown_for_actions` option and disable tracking under **WooCommerce > Settings > Advanced > Woocommerce.com**
9. Make sure the notice doesn't show up for the above anymore and that `customer-effort-score-exit-page` in local storage does not get updated, but stays empty.



### Other information:




### FOR PR REVIEWER ONLY:
